flow/record/adapter/elastic.py

```diff
@@ -87,25 +104,33 @@ def flush(self) -> None:
         pass
 
     def close(self) -> None:
-        self.queue.put(StopIteration)
-        self.event.wait()
-        self.es.close()
+        if hasattr(self, "es"):
```

Comment:
If `elasticsearch.Elasticsearch()` can indeed return `None`, we should just raise an exception in`__init__()`. Do you have an example uri that would make `self.es = None`?

Reply:
The `elasticsearch.Elasticsearch()` does not return `None`, but raise an ValueError with invalid scheme. The deconstructor function of the class then calls the `close()` function, which contains function calls on undefined variables. See the comment for the stack strace.

Reply:
This is also why variables are defined in other adapters as well. Such as the `splunk.py` where the `self.sock` is set. Its also possible to use a class variable inside this file, but I prefer the way of `hasattr` above the class variables.

Reply:
gotcha!